tests/unit/lambdas/daemons/test_worker.py

```diff
@@ -0,0 +1,93 @@
+import os
+import uuid
+
+import boto3
+from unittest import mock
+
+from ... import MatrixTestCaseUsingMockAWS
+from ... import test_bundle_spec
+from matrix.lambdas.daemons.worker import Worker
+from matrix.common.dynamo_handler import DynamoHandler
+from matrix.common.dynamo_handler import StateTableField
+from matrix.common.dynamo_handler import DynamoTable
+from matrix.common.lambda_handler import LambdaName
+
+
+class TestWorker(MatrixTestCaseUsingMockAWS):
+
+    def setUp(self):
+        super(TestWorker, self).setUp()
+
+        self.dynamo = boto3.resource("dynamodb", region_name=os.environ['AWS_DEFAULT_REGION'])
+        self.state_table_name = os.environ['DYNAMO_STATE_TABLE_NAME']
+        self.output_table_name = os.environ['DYNAMO_OUTPUT_TABLE_NAME']
+        self.request_id = str(uuid.uuid4())
+        self.create_test_state_table(self.dynamo)
+        self.create_test_output_table(self.dynamo)
+        self.handler = DynamoHandler()
+        self.format_string = "zarr"
+        self.worker_chunk_spec = {
+            "bundle_uuid": test_bundle_spec['uuid'],
+            "bundle_version": test_bundle_spec['version'],
+            "start_row": 100,
+            "num_rows": 6500
+        }
+        self.worker = Worker(self.request_id)
+
+    @mock.patch('matrix.common.s3_zarr_store.S3ZarrStore.write_from_pandas_dfs')
+    @mock.patch('matrix.common.dss_zarr_store.DSSZarrStore.__init__')
+    @mock.patch('zarr.group')
+    @mock.patch('matrix.lambdas.daemons.worker.convert_dss_zarr_root_to_subset_pandas_dfs')
+    @mock.patch('matrix.lambdas.daemons.worker.Worker._check_if_all_workers_and_mappers_for_request_are_complete')
+    @mock.patch('matrix.common.dynamo_handler.DynamoHandler.increment_table_field')
+    @mock.patch('matrix.common.s3_zarr_store.S3ZarrStore.write_column_data')
+    @mock.patch("matrix.common.lambda_handler.LambdaHandler.invoke")
+    def test_run(self, mock_lambda_handler_invoke, mock_write_column_data, mock_increment_dynamo, mock_is_complete,
+                 mock_df_conversion, mock_zarr_group, mock_dss_zarr_store, mock_write_to_s3):
+        mock_dss_zarr_store.return_value = None
+        mock_write_to_s3.return_value = None
+        mock_zarr_group.return_value = None
+        mock_df_conversion.return_value = (None, None)
+        mock_is_complete.return_value = True
+        self.worker.run(self.format_string, self.worker_chunk_spec)
+        expected_calls = [
+            mock.call(LambdaName.REDUCER, {'request_id': self.request_id,
+                                           'format': self.format_string}),
+        ]
+        self.assertEqual(mock_lambda_handler_invoke.call_count, 1)
+        mock_lambda_handler_invoke.assert_has_calls(expected_calls)
```

Comment:
For asserting single calls, this works too:
`mock_lambda_handler.assert_called_once_with(LambdaName.REDUCER, {'request_id': self.request_id, 'format': self.format_string})`